### PR TITLE
Sc 16865

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

Updates topic validation method to use TopicNameRegex. The following characters have been escaped within the expression: `.` `-` `_`. 

This also adds a length check to the topic name. (This has been added on the frontend, but was not previously done on the backend.)

Fixes SC-16865

### Type of change

- [ ] new feature
- [x] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/db/errors.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ var (
 
 	// Max-length errors
 	ErrProjectDescriptionTooLong = errors.New("project description is too long")
+	ErrTopicNameTooLong          = errors.New("topic name is too long")
```
Comment (pdeziel) [inline pkg/tenant/db/errors.go:48]: Can you make sure the frontend is validating this as well or create a story for it?

File: pkg/tenant/db/topics_test.go [hunk 2]
```diff
@@ -72,8 +73,12 @@ func TestTopicValidate(t *testing.T) {
 	topic.Name = "otters;"
 	require.ErrorIs(t, topic.Validate(), db.ErrInvalidTopicName, "expected invalid name to be an error")
 
+	// Test name that is too long
+	topic.Name = strings.Repeat("a", 513)
+	require.ErrorIs(t, topic.Validate(), db.ErrTopicNameTooLong, "expected too long name to be an error")
```
Comment (pdeziel) [inline pkg/tenant/db/topics_test.go:78]: Thanks for adding the test case!

File: pkg/tenant/db/topics_test.go [hunk 3]
```diff
@@ -72,8 +73,12 @@ func TestTopicValidate(t *testing.T) {
...
+
 	// Valid topic
-	topic.Name = "otters"
+	topic.Name = "otters_are-cool"
```
Comment (pdeziel) [inline pkg/tenant/db/topics_test.go:81]: Since the validation is a bit complicated, I think we should use a test case pattern similar to the following.

```golang
testCases := struct{}{
    name string,
    err error
}{
    ("otters;", db.ErrInvalidTopicName},
    ("otters_are-cool", nil),
}

for i, tc := testCases {
    if tc.err == nil {
        require.NoError(topic.Validate(), fmt.Sprintf("expected no error for test case %d", i)
    } else {
        require.ErrorIs(tc.err, topic.Validate(), fmt.Sprintf("expected error for test case %d", i)
    }
}
```

That will allows us to keep track of all the edge cases, so can you add a story for a future sprint to revise the test and add some more edge cases?

**Review by pdeziel — APPROVED** [review]
Looks great, thanks!

File: pkg/tenant/db/errors.go (hunk 1 above)
Comment (daniellemaxwell) [reply pkg/tenant/db/errors.go:48]: I'll double check the regex currently in use for validation on the FE side.
